include/zarray/zreducer_options.hpp

```diff
@@ -0,0 +1,266 @@
+#ifndef XTENSOR_ZREDUCER_OPTIONS_HPP
+#define XTENSOR_ZREDUCER_OPTIONS_HPP
+
+#include "xtensor/xcontainer.hpp"
+#include "xtensor/xscalar.hpp"
+#include "xtensor/xreducer.hpp"
+
+
+namespace xt
+{
+    template <class CTE>
+    class zscalar_wrapper;
+
+    template <class F, class CT>
+    class zreducer;
+
+    class zreducer_options{
+    public:
+
+
+        template <class X>
+        struct initial_tester : std::false_type {};
+
+        template <class X>
+        struct initial_tester<xinitial<X>> : std::true_type {};
+
+        // Workaround for Apple because tuple_cat is buggy!
+        template <class X>
+        struct initial_tester<const xinitial<X>> : std::true_type {};
+
+
+
+        template <class F, class CT>
+        friend class zreducer;
+
+    private:
+        template<class T>
+        using make_scalar_wrapper_t = zscalar_wrapper<xscalar<T>>;
```

Comment:
Naming; usually, `make_xxx` are used for builder functions, I think `scalar_wrapper_type` would be fine here.